reporter/src/main/kotlin/reporters/TableReporter.kt

```diff
@@ -235,11 +235,11 @@ abstract class TableReporter : Reporter() {
             val tableRows = (listOf(project.id) + project.collectDependencyIds()).map { id ->
                 val scanResult = scanRecord.scanResults.find { it.id == id }
 
-                val scopes = project.scopes.filter { id in it }.associate {
-                    val scopeExcludes = ortResult.repository.config.excludes?.findScopeExcludes(it, project)
-                            ?: emptyList()
-                    Pair(it.name, scopeExcludes)
-                }.toSortedMap()
+                val scopes = project.scopes.filter { id in it }.let { scopes ->
+                    ortResult.repository.config.excludes
+                            ?.scopeExcludesByName(project, scopes)?.toSortedMap()
+                            ?: scopes.associateTo(sortedMapOf()) { Pair(it.name, emptyList<ScopeExclude>()) }
```

Comment:
To be honest, I found the original code to be more readable. It seems a bit clumsy that we have to create a whole new map in case there are no `excludes` now.

How about instead adding a function to `Project` that takes an `Exclude` and is called something like `getExcludedScopes`?

Reply:
I agree that the code in TableReporter did not become more readable, but I want to bundle all exclude related logic in functions in the exclude classes, to have it not scattered around, and to make it testable. The reporter code needs to be cleaned up separately.

Reply:
Ok.